api/app/signals/apps/my_signals/rest_framework/views/signals.py

```diff
@@ -0,0 +1,86 @@
+# SPDX-License-Identifier: MPL-2.0
+# Copyright (C) 2022 Gemeente Amsterdam
+from datapunt_api.pagination import HALPagination
+from datapunt_api.rest import DEFAULT_RENDERERS
+from dateutil.relativedelta import relativedelta
+from django.conf import settings
+from django.http import Http404
+from django.utils import timezone
+from django_filters.rest_framework import DjangoFilterBackend
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from rest_framework.status import HTTP_501_NOT_IMPLEMENTED
+from rest_framework.viewsets import ReadOnlyModelViewSet
+from rest_framework_extensions.mixins import DetailSerializerMixin
+
+from signals.apps.api.generics.filters import FieldMappingOrderingFilter
+from signals.apps.my_signals.rest_framework.filters.signals import MySignalFilterSet
+from signals.apps.my_signals.rest_framework.serializers.signals import (
+    SignalDetailSerializer,
+    SignalListSerializer
+)
+from signals.apps.signals.models import Signal
+
+
+class MySignalsViewSet(DetailSerializerMixin, ReadOnlyModelViewSet):
+    renderer_classes = DEFAULT_RENDERERS
+    pagination_class = HALPagination
+
+    # TODO: SIG-4757 add login/authentication flow
+    authentication_classes = ()
+
+    lookup_field = 'uuid'
+    lookup_url_kwarg = 'uuid'
+
+    serializer_class = SignalListSerializer
+    serializer_detail_class = SignalDetailSerializer
+
+    filter_backends = (DjangoFilterBackend, FieldMappingOrderingFilter,)
+    filterset_class = MySignalFilterSet
+
+    ordering = (
+        '-created_at',
+    )
+    ordering_fields = (
+        'created_at',
+    )
+    ordering_field_mappings = {
+        'created_at': 'created_at',
+    }
+
+    def get_queryset(self, *args, **kwargs):
+        """
+        Only return signals that are in a specific state and created in the last year
+
+        TODO: SIG-4757 add login/authentication flow
+        """
+        one_year_ago = timezone.now() - relativedelta(years=1)
+
+        return Signal.objects.filter(
```

Comment:
This QuerySet needs some kind of filtering for the reporter requesting the Signal instances, right? If we have a list endpoint we must filter this to only show complaints/Signals tied to a specific email address (as that was how we tie a reporter to his or her own complaints). For a detail endpoint that is not as much of a problem because we can know which complaint/Signal to show based on the requested UUID.

I think what is missing from this class is not just the Authentication class but also some filtering for logged in Reporter.

Reply:
https://github.com/Amsterdam/signals/pull/1093